src/main/java/edu/upf/taln/dri/lib/model/util/ObjectGenerator.java

```diff
@@ -1600,307 +1599,17 @@ else if(titleLang.trim().toLowerCase().equals("es")) {
-													}
-												}
-												
-												// Country
-												List<Annotation> countryList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterJATS.JATSannSet, ImporterJATS.JATSaffiliationAddressLine_STATE, affil);
-												if(countryList != null && countryList.size() > 0) {
-													Optional<String> countryOpt = GateUtil.getAnnotationText(countryList.get(0), cacheManager.getGateDoc());
-													if(countryOpt.isPresent() && countryOpt.get() != null && !countryOpt.get().equals("")) {
-														newAffiliation.setState(normalizeText(countryOpt.get()).trim());
-													}
-												}
-												
-												// URL / ext-link
-												List<Annotation> uriList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterJATS.JATSannSet, ImporterJATS.JATSaffiliationAddressLine_URI, affil);
-												List<Annotation> extLinkList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterJATS.JATSannSet, ImporterJATS.JATSaffiliationAddressLine_EXTLINK, affil);
-												if(uriList != null && uriList.size() > 0) {
-													Optional<String> uriOpt = GateUtil.getAnnotationText(uriList.get(0), cacheManager.getGateDoc());
-													if(uriOpt.isPresent() && uriOpt.get() != null && !uriOpt.get().equals("")) {
-														newAffiliation.setURL(normalizeText(uriOpt.get()).trim());
-													}
-												}
-												else if(extLinkList != null && extLinkList.size() > 0) {
-													Optional<String> extLinkOpt = GateUtil.getAnnotationText(extLinkList.get(0), cacheManager.getGateDoc());
-													if(extLinkOpt.isPresent() && extLinkOpt.get() != null && !extLinkOpt.get().equals("")) {
-														newAffiliation.setURL(normalizeText(extLinkOpt.get()).trim());
-													}
-												}
-												
-												// Email of author
-												List<Annotation> emailList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterJATS.JATSannSet, ImporterJATS.JATSaffiliationAddressLine_EMAIL, affil);
-												if(emailList != null && emailList.size() > 0) {
-													Optional<String> emailOpt = GateUtil.getAnnotationText(emailList.get(0), cacheManager.getGateDoc());
-													if(emailOpt.isPresent() && emailOpt.get() != null && !emailOpt.get().equals("")) {
-														newAuthor.setEmail(normalizeText(emailOpt.get()).trim());
-													}
-												}
-												
-												if(newAffiliation.getName() != null && !newAffiliation.getName().equals("")) {
-													newAuthor.addAffiliation(newAffiliation);
-												}
-												
-											}
-										}
-									}
-									
-									resultImpl.addAuthor(newAuthor);
-								}
-							}
-						}
-					}
-					
+				catch (Exception e) {
+					System.out.println("Could not use an Extractor for the authors");
 				}
-				if(sourceDocFeature != null && sourceDocFeature.equals(SourceENUM.PDFEXT.toString())) {
```

Comment:
We note that instead of also being an `if-else` statement like JATS, this is its own `if` statement. We believe this is an error because `sourceDocFeature` wouldn't be able to be both equal to GROBID/JATS and PDFEXT at the same time.

As such, we treat it just like an if-else statement and extract it into `PDFEXTAuthorExtractor` which is picked by the `ExtractorFactory`.

We leave it to `ObjectGenerator` to convert `sourceDocFeature` into the enum. If it fails (or the extractors fail) an exception is caught here and should continue normal execution without author extraction.